llvm-libgcc/CMakeLists.txt

```diff
@@ -1,19 +1,26 @@
-cmake_minimum_required(VERSION 3.20.0)
+#===============================================================================
+# Setup Project
+#===============================================================================
 
-if (NOT IS_DIRECTORY "${CMAKE_CURRENT_LIST_DIR}/../libunwind")
-  message(FATAL_ERROR "llvm-libgcc requires being built in a monorepo layout with libunwind available")
-endif()
+cmake_minimum_required(VERSION 3.20.0)
 
 set(LLVM_COMMON_CMAKE_UTILS "${CMAKE_CURRENT_SOURCE_DIR}/../cmake")
 
-list(APPEND CMAKE_MODULE_PATH
+# Add path for custom modules
+list(INSERT CMAKE_MODULE_PATH 0
   "${CMAKE_CURRENT_SOURCE_DIR}/cmake"
   "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules"
+  "${CMAKE_CURRENT_SOURCE_DIR}/../runtimes/cmake/Modules"
   "${LLVM_COMMON_CMAKE_UTILS}"
   "${LLVM_COMMON_CMAKE_UTILS}/Modules"
   )
 
-project(llvm-libgcc LANGUAGES C CXX ASM)
```

Comment:
Why is this being removed?

Reply:
`libunwind/CMakeLists.txt` does not contains a `project(libunwind LANGUAGES C CXX ASM)`. To keep consistent with `libunwind/CMakeLists.txt`, `project(llvm-libgcc LANGUAGES C CXX ASM)` is not added.

Reply:
Now `llvm-libgcc` use the same approach as `compiler-rt` in [60eccd0](https://github.com/llvm/llvm-project/pull/65455/commits/60eccd0b5b012e3e13bfdd4145dbd381a955887b).

Reply:
I'm not following what you mean, sorry. Since this is the driver for compiler-rt and libunwind, it makes sense to me for this project to indicate this information.

Reply:
`compiler-rt/CMakeLists.txt` detects whether `compiler-rt` is built as a subproject of `runtimes` or a standalone project. In standalone mode it sets options to detect `llvm` and declares the project name with `project(CompilerRT C CXX ASM)`.

Now `llvm-libgcc/CMakeLists.txt` does the same: in standalone mode, it tells added `compiler-rt` that we are not built as a subproject of `runtimes` and we have to detect `llvm` ourselves.

Here "standalone" means `cmake -B build -S compiler-rt` or `cmake -B build -S llvm-libgcc`, not `cmake -B build -S runtimes` nor `cmake -B build -S llvm`. Even in the whole `llvm-project` source tree, directly using `compiler-rt/CMakeLists.txt` or `llvm-libgcc/CMakeLists.txt` is "standalone".

If included by other `CMakeLists.txt`, `runtimes/CMakeLists.txt` for example, it does not matter whether there is a `project(CompilerRT C CXX ASM)` or `project(llvm-libgcc LANGUAGES C CXX ASM)` declaration actually performed. To keep consistent with `compiler-rt`, `llvm-libgcc/CMakeLists.txt` is adjusted correspondingly in [60eccd0](https://github.com/llvm/llvm-project/pull/65455/commits/60eccd0b5b012e3e13bfdd4145dbd381a955887b).

Correspondingly lines from [`compiler-rt/CMakeLists.txt`](https://github.com/llvm/llvm-project/blob/60eccd0b5b012e3e13bfdd4145dbd381a955887b/compiler-rt/CMakeLists.txt#L12) :
``` CMake
# Check if compiler-rt is built as a standalone project.
if (CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR OR COMPILER_RT_STANDALONE_BUILD)
  project(CompilerRT C CXX ASM)
  set(COMPILER_RT_STANDALONE_BUILD TRUE)
  set_property(GLOBAL PROPERTY USE_FOLDERS ON)
endif()
```

Correspondingly lines in [`llvm-libgcc/CMakeLists.txt`](https://github.com/llvm/llvm-project/blob/60eccd0b5b012e3e13bfdd4145dbd381a955887b/llvm-libgcc/CMakeLists.txt#L9):
``` CMake
# Check if llvm-libgcc is built as a standalone project
if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR OR LLVM_LIBGCC_STANDALONE_BUILD)
  project(llvm-libgcc LANGUAGES C CXX ASM)
  set(COMPILER_RT_STANDALONE_BUILD ON)
  set_property(GLOBAL PROPERTY USE_FOLDERS ON)
  set(LLVM_LIBGCC_COMPILER_RT_BINARY_DIR "compiler-rt")
  set(LLVM_LIBGCC_LIBUNWIND_BINARY_DIR "libunwind")
else()
  set(LLVM_LIBGCC_COMPILER_RT_BINARY_DIR "../compiler-rt")
  set(LLVM_LIBGCC_LIBUNWIND_BINARY_DIR "../libunwind")
endif()
```

And I am a bit confused how should we indicate "this is the driver for `compiler-rt` and `libunwind`". It is completely acceptable to perform `project(llvm-libgcc LANGUAGES C CXX ASM)` declaration unconditionally, but does it makes sense in this scenario?

Reply:
Understood, thanks for highlighting the change to [llvm-libgcc/CMakeLists.txt](https://github.com/llvm/llvm-project/blob/60eccd0b5b012e3e13bfdd4145dbd381a955887b/llvm-libgcc/CMakeLists.txt#L9), which I'd missed.